### PR TITLE
fix: Drop carousel bug on reload

### LINKED ISSUE #10005: Drop carousel bug on reload
https://github.com/kodadot/nft-gallery/assets/90852205/9879b6e9-1b93-4409-9799-42e37ad31072

When you enter the page, currently there is this big card that scales to the width of the screen and then comes back

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #10005

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2024-04-06 at 14 03 02

### CODE REVIEW

**Comment by vikiival** [comment]
Ready for review?

File: components/carousel/module/CarouselAgnostic.vue [hunk 1]
```diff
@@ -142,15 +148,11 @@ if (props.config) {
...
-  watchDebounced(
-    [() => props.config, () => props.items.length],
-    () => {
-      if (slider.value) {
-        slider.value.update(props.config)
-      }
-    },
-    { debounce: SYNC_CONFIG_DEBOUNCE_AMOUNT, immediate: true },
-  )
+  onMounted(updateConfig)
```
Comment (daiagi) [inline components/carousel/module/CarouselAgnostic.vue:151]: is there a diff between `immeditate: true` and `onMounted` here?

File: components/carousel/module/CarouselAgnostic.vue [hunk 2]
```diff
@@ -142,15 +148,11 @@ if (props.config) {
...
+
+  watchDebounced([() => props.config, () => props.items.length], updateConfig, {
+    debounce: SYNC_CONFIG_DEBOUNCE_AMOUNT,
+  })
```
Comment (daiagi) [inline components/carousel/module/CarouselAgnostic.vue:155]: simpler?

```  suggestion
watchDebounced([() => props.config, () => props.items.length], () => slider.value?.update(props.config), {
    debounce: SYNC_CONFIG_DEBOUNCE_AMOUNT,
  })
  
  ```
  
  
  and delete `updateConfig` function

Comment (hassnian) [reply components/carousel/module/CarouselAgnostic.vue]: yep, done

File: composables/useDynamicGrid.ts [hunk 3]
```diff
@@ -46,10 +46,14 @@ export default ({
     return fillRows ? getColsFilledByAllRows(getCols) : getCols
   })
 
+  const isReady = computed(() => Boolean(containerWidth.value))
+
+  onMounted(() => (containerWidth.value = container.value?.clientWidth ?? 0))
```
[the remark this answers is not in the document]
Comment (hassnian) [reply composables/useDynamicGrid.ts:51]: child's on mount gets triggered before this one

**Review by vikiival — APPROVED** [review]

**Review by daiagi — APPROVED** [review]

**Pull request merged**